ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/PrintPublisher.java

```diff
@@ -0,0 +1,130 @@
+/**
+ * Copyright 2018 Confluent Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ **/
+
+package io.confluent.ksql.rest.server.resources.streaming;
+
+import com.google.common.util.concurrent.ListeningScheduledExecutorService;
+
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+import org.apache.kafka.common.TopicPartition;
+import org.apache.kafka.common.serialization.BytesDeserializer;
+import org.apache.kafka.common.serialization.StringDeserializer;
+import org.apache.kafka.common.utils.Bytes;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.ksql.rest.server.resources.streaming.Flow.Subscriber;
+import io.confluent.ksql.rest.server.resources.streaming.TopicStream.RecordFormatter;
+
+public class PrintPublisher implements Flow.Publisher<Collection<String>> {
```

Comment:
Javadocs for this class and the `StreamPublisher` would help: what are the implementations supposed to do? how do they differ? etc. I know one supports `print topic` and the other supports `select * ..`, but it is not all that obvious.

Reply:
sure I can add some docs, I thought it would be pretty explicit from the instanceof checks at https://github.com/confluentinc/ksql/pull/1285/files#diff-7b2147b0c6a02c24d6f558dbca9b78aaR116 and https://github.com/confluentinc/ksql/pull/1285/files#diff-7b2147b0c6a02c24d6f558dbca9b78aaR126